components/UserProvider.js

```diff
@@ -94,9 +97,21 @@ class UserProvider extends React.Component {
       });
       return LoggedInUser;
     } catch (error) {
-      // If token from localStorage is invalid or expired, delete it
-      if (!token && ['Invalid token', 'Expired token'].includes(error.message)) {
+      // Malformed tokens are detected and removed by the frontend in `lib/hooks/withLoggedInUser.js` (search for "malformed")
+      // Invalid tokens are ignored in the API, the user is treated as unauthenticated (see `parseJwt` in `server/middleware/authentication.js`)
+      // There can therefore only be two types of errors here:
+      // - Network/server errors: we'll display a message
+      // - Expired tokens: we'll logout the user with a "Your session has expired. Please sign-in again." message
+      const errorType = get(error, 'networkError.result.error.type');
+
+      // For expired tokens, we directly logout & show a toast as we want to make sure it gets
+      // displayed not matter what page the user is on.
+      if (!token && errorType === 'jwt_expired') {
```

Comment:
This is the root fix: the error here was not retuned with the "Expired token" message anymore, but rather looks like:

```json
{"graphQLErrors":[],"networkError":{"name":"ServerError","response":{},"statusCode":401,"result":{"error":{"code":401,"type":"jwt_expired","message":"jwt expired"}}},"message":"Response not successful: Received status code 401"}
```